# 0.데이터 불러오기

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

!pip install koreanize-matplotlib
import koreanize_matplotlib

In [3]:
# pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

In [5]:
p_unique = pd.read_csv('/content/drive/MyDrive/최종프로젝트/merge2_onlyC.csv')
cont = pd.read_csv('/content/drive/MyDrive/최종프로젝트/cont_merge_ver2_(drop).csv')
pmnt_pol = pd.read_csv('/content/drive/MyDrive/최종프로젝트/pmntdtl_pol.csv')

# 1.차주정보 + 계약정보 결합

p_unique + cont 두 데이터 결합 시 58429개 행 살아남음
  * 결합된 테이블 : merge_p_cont

In [11]:
p_unique.sort_values('POL_SN')

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP
52849   902434        8              C       1   29          2
40794   996067       37              C       1   73          7
69823   930108       43              C       1   43          4
29047  1891958       87              C       2   79          7
60137  1667175      111              C       2   42          4
...        ...      ...            ...     ...  ...        ...
61635  1103597  9997220              C       2   40          4
33655  1325341  9997268              C       2   40          4
59050  1494449  9997274              C       2   55          5
41298  1052214  9997340              C       1   23          2
34414   751928  9997367              C       1   44          4

[129442 rows x 6 columns]

In [15]:
cont.sort_values('POL_SN')

POL_SN  reduce_GBN_CD  reduce_STAT_CD  YM_202006  CT_CHNL_CD  \
70660         8              5               1          1           1   
0           111              1               1          1           3   
1           179              1               1          1           1   
2           311              3               1          1           4   
3           478              4               1          1           1   
...         ...            ...             ...        ...         ...   
70645   9997129              3               1          1           1   
110842  9997240              6               4          1           3   
70646   9997274              2               1          1           6   
70647   9997286              2               1          1           6   
110843  9997320              4               2          1           1   

        CT_PY_AMT  CT_PY_CYCLE_CD  CT_PY_PD  mean_AMT  month_AMT   z_score  \
70660          35               2        15      35.0       35.0 -0.036977   
0              60               2        25      60.0       60.0 -0.030098   
1             330               2        20     330.0      330.0  0.044197   
2             120               3         1     120.0       10.0 -0.013588   
3             110               2        20     110.0      110.0 -0.016339   
...           ...             ...       ...       ...        ...       ...   
70645          24               2         5      24.0       24.0 -0.040004   
110842         24               1         0      24.0       24.0 -0.040004   
70646          34               2        20      34.0       34.0 -0.037252   
70647          33               2        20      33.0       33.0 -0.037527   
110843         81               2        30      81.0       81.0 -0.024319   

        m_z_score  GBN_CD_1  GBN_CD_2  GBN_CD_3  GBN_CD_4  
70660   -0.035377         1         1         1         1  
0       -0.028465         1         1         1         0  
1        0.046178         1         1         1         0  
2       -0.042288         1         1         1         0  
3       -0.014643         1         1         1         1  
...           ...       ...       ...       ...       ...  
70645   -0.038418         1         1         0         1  
110842  -0.038418         1         0         0         0  
70646   -0.035653         0         1         0         0  
70647   -0.035930         0         1         0         0  
110843  -0.022660         1         1         1         1  

[110844 rows x 16 columns]

In [13]:
merge_p_cont = pd.merge(p_unique, cont, on='POL_SN', how='inner').sort_values('POL_SN')
merge_p_cont

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
23752   902434        8              C       1   29          2              5   
27041  1667175      111              C       2   42          4              1   
22999   594950      179              C       1   45          4              1   
4257   1388667      722              C       2   20          2              3   
44166   453567     1031              C       2   54          5              2   
...        ...      ...            ...     ...  ...        ...            ...   
43155   242990  9996700              C       2   30          3              1   
36539  1697168  9996939              C       1   23          2              3   
14597   314074  9997065              C       1   39          3              1   
40114  1406609  9997129              C       2   46          4              3   
26549  1494449  9997274              C       2   55          5              2   

       reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
23752               1          1           1         35               2   
27041               1          1           3         60               2   
22999               1          1           1        330               2   
4257                1          1           1         28               2   
44166               1          1           6         29               2   
...               ...        ...         ...        ...             ...   
43155               1          1           1         47               2   
36539               1          1           6         25               2   
14597               1          1           1        190               2   
40114               1          1           1         24               2   
26549               1          1           6         34               2   

       CT_PY_PD    mean_AMT   month_AMT   z_score  m_z_score  GBN_CD_1  \
23752        15   35.000000   35.000000 -0.036977  -0.035377         1   
27041        25   60.000000   60.000000 -0.030098  -0.028465         1   
22999        20  330.000000  330.000000  0.044197   0.046178         1   
4257         30   28.000000   28.000000 -0.038903  -0.037312         1   
44166        20   29.000000   29.000000 -0.038628  -0.037035         0   
...         ...         ...         ...       ...        ...       ...   
43155        20   45.666667   45.666667 -0.034042  -0.032428         1   
36539        25   25.000000   25.000000 -0.039728  -0.038141         1   
14597        20  190.000000  190.000000  0.005674   0.007474         1   
40114         5   24.000000   24.000000 -0.040004  -0.038418         1   
26549        20   34.000000   34.000000 -0.037252  -0.035653         0   

       GBN_CD_2  GBN_CD_3  GBN_CD_4  
23752         1         1         1  
27041         1         1         0  
22999         1         1         0  
4257          1         1         1  
44166         1         1         0  
...         ...       ...       ...  
43155         1         1         0  
36539         1         0         0  
14597         1         1         0  
40114         1         0         1  
26549         1         0         0  

[58429 rows x 21 columns]

In [10]:
cont[cont['POL_SN'] == 1486482]

Empty DataFrame
Columns: [POL_SN, reduce_GBN_CD, reduce_STAT_CD, YM_202006, CT_CHNL_CD, CT_PY_AMT, CT_PY_CYCLE_CD, CT_PY_PD, mean_AMT, month_AMT, z_score, m_z_score, GBN_CD_1, GBN_CD_2, GBN_CD_3, GBN_CD_4]
Index: []

In [14]:
cont[cont['POL_SN'] == 8]

POL_SN  reduce_GBN_CD  reduce_STAT_CD  YM_202006  CT_CHNL_CD  \
70660       8              5               1          1           1   

       CT_PY_AMT  CT_PY_CYCLE_CD  CT_PY_PD  mean_AMT  month_AMT   z_score  \
70660         35               2        15      35.0       35.0 -0.036977   

       m_z_score  GBN_CD_1  GBN_CD_2  GBN_CD_3  GBN_CD_4  
70660  -0.035377         1         1         1         1

In [16]:
p_unique[p_unique['POL_SN'] == 478]

Empty DataFrame
Columns: [JOIN_SN, POL_SN, CT_RLPS_GBN_CD, GENDER, AGE, AGE_GROUP]
Index: []

In [26]:
cont[cont.duplicated(subset=['POL_SN'], keep=False)]

POL_SN  reduce_GBN_CD  reduce_STAT_CD  YM_202006  CT_CHNL_CD  \
849      118848              4               1          1           1   
850      118848              4               1          1           1   
4997     711858              4               1          1           1   
4998     711858              4               1          1           1   
11663   1660417              3               1          1           1   
11664   1660417              3               1          1           1   
14462   2047731              6               1          1           2   
14463   2047731              6               1          1           2   
17267   2431715              6               4          1           2   
17268   2431715              6               4          1           2   
18906   2666864              6               2          1           2   
18907   2666864              6               2          1           2   
20152   2838555              6               4          1           2   
20153   2838555              6               4          1           2   
24137   3409431              3               1          1           7   
24138   3409431              3               1          1           7   
26012   3678151              4               1          1           1   
26013   3678151              4               1          1           1   
31907   4504290              6               2          1           5   
31908   4504290              6               2          1           5   
41401   5857154              3               1          1           3   
41402   5857154              3               1          1           3   
64347   9108814              4               1          1           1   
64348   9108814              4               1          1           1   
67440   9533704              3               4          1           7   
67441   9533704              3               4          1           7   
67442   9533704              3               4          1           7   
71991    308355              6               4          1           2   
71992    308355              6               4          1           2   
75033   1045082              3               4          1           1   
75034   1045082              3               4          1           1   
88271   4372529              4               1          1           3   
88272   4372529              4               1          1           3   
101366  7635944              6               4          1           7   
101367  7635944              6               4          1           7   
101409  7644968              6               4          1           2   
101410  7644968              6               4          1           2   
106057  8815494              6               4          1           2   
106058  8815494              6               4          1           2   
106749  8984788              6               4          1           2   
106750  8984788              6               4          1           2   

        CT_PY_AMT  CT_PY_CYCLE_CD  CT_PY_PD  mean_AMT  month_AMT   z_score  \
849            36               2        20      36.0       36.0 -0.036702   
850            36               2        20      36.0       36.0 -0.036702   
4997           38               2        15      38.0       38.0 -0.036151   
4998           38               2        15      38.0       38.0 -0.036151   
11663        1000               1         0    1000.0     1000.0  0.228556   
11664        1000               1         0    1000.0     1000.0  0.228556   
14462          26               1         0      26.0       26.0 -0.039453   
14463          26               1         0      26.0       26.0 -0.039453   
17267          13               1         0      13.0       13.0 -0.043030   
17268          13               1         0      13.0       13.0 -0.043030   
18906           1               1         0       1.0        1.0 -0.046332   
18907 

In [27]:
merge_p_cont[merge_p_cont['POL_SN'] == 7644968]

Empty DataFrame
Columns: [JOIN_SN, POL_SN, CT_RLPS_GBN_CD, GENDER, AGE, AGE_GROUP, reduce_GBN_CD, reduce_STAT_CD, YM_202006, CT_CHNL_CD, CT_PY_AMT, CT_PY_CYCLE_CD, CT_PY_PD, mean_AMT, month_AMT, z_score, m_z_score, GBN_CD_1, GBN_CD_2, GBN_CD_3, GBN_CD_4]
Index: []

In [29]:
p_unique[p_unique['POL_SN'] == 7644968]

Empty DataFrame
Columns: [JOIN_SN, POL_SN, CT_RLPS_GBN_CD, GENDER, AGE, AGE_GROUP]
Index: []

In [28]:
merge_p_cont[merge_p_cont.duplicated(subset=['POL_SN'], keep=False)]

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
11670   263297  9108814              C       2   37          3              4   
11669   263297  9108814              C       2   37          3              4   

       reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
11670               1          1           1         73               2   
11669               1          1           1         73               2   

       CT_PY_PD  mean_AMT  month_AMT   z_score  m_z_score  GBN_CD_1  GBN_CD_2  \
11670        20      73.0       73.0 -0.026521  -0.024871         1         0   
11669        20      73.0       73.0 -0.026521  -0.024871         1         1   

       GBN_CD_3  GBN_CD_4  
11670         0         0  
11669         1         1

* 중복되는 데이터. 날 데이터 봤을 때,
  * 인덱스번호 11670은 263297 차주건
  * 인덱스번호 11669은 128420 차주건
어차피 중복되는 데이터 날리면 될 듯

In [31]:
# 위의 중복 건 제거
merge_p_cont.drop_duplicates(subset=['POL_SN'], keep=False, inplace=True)
merge_p_cont

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
23752   902434        8              C       1   29          2              5   
27041  1667175      111              C       2   42          4              1   
22999   594950      179              C       1   45          4              1   
4257   1388667      722              C       2   20          2              3   
44166   453567     1031              C       2   54          5              2   
...        ...      ...            ...     ...  ...        ...            ...   
43155   242990  9996700              C       2   30          3              1   
36539  1697168  9996939              C       1   23          2              3   
14597   314074  9997065              C       1   39          3              1   
40114  1406609  9997129              C       2   46          4              3   
26549  1494449  9997274              C       2   55          5              2   

       reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
23752               1          1           1         35               2   
27041               1          1           3         60               2   
22999               1          1           1        330               2   
4257                1          1           1         28               2   
44166               1          1           6         29               2   
...               ...        ...         ...        ...             ...   
43155               1          1           1         47               2   
36539               1          1           6         25               2   
14597               1          1           1        190               2   
40114               1          1           1         24               2   
26549               1          1           6         34               2   

       CT_PY_PD    mean_AMT   month_AMT   z_score  m_z_score  GBN_CD_1  \
23752        15   35.000000   35.000000 -0.036977  -0.035377         1   
27041        25   60.000000   60.000000 -0.030098  -0.028465         1   
22999        20  330.000000  330.000000  0.044197   0.046178         1   
4257         30   28.000000   28.000000 -0.038903  -0.037312         1   
44166        20   29.000000   29.000000 -0.038628  -0.037035         0   
...         ...         ...         ...       ...        ...       ...   
43155        20   45.666667   45.666667 -0.034042  -0.032428         1   
36539        25   25.000000   25.000000 -0.039728  -0.038141         1   
14597        20  190.000000  190.000000  0.005674   0.007474         1   
40114         5   24.000000   24.000000 -0.040004  -0.038418         1   
26549        20   34.000000   34.000000 -0.037252  -0.035653         0   

       GBN_CD_2  GBN_CD_3  GBN_CD_4  
23752         1         1         1  
27041         1         1         0  
22999         1         1         0  
4257          1         1         1  
44166         1         1         0  
...         ...       ...       ...  
43155         1         1         0  
36539         1         0         0  
14597         1         1         0  
40114         1         0         1  
26549         1         0         0  

[58427 rows x 21 columns]

In [32]:
merge_p_cont[merge_p_cont.duplicated(subset=['POL_SN'], keep=False)]

Empty DataFrame
Columns: [JOIN_SN, POL_SN, CT_RLPS_GBN_CD, GENDER, AGE, AGE_GROUP, reduce_GBN_CD, reduce_STAT_CD, YM_202006, CT_CHNL_CD, CT_PY_AMT, CT_PY_CYCLE_CD, CT_PY_PD, mean_AMT, month_AMT, z_score, m_z_score, GBN_CD_1, GBN_CD_2, GBN_CD_3, GBN_CD_4]
Index: []

# 2.보험금 지급 내용까지 결합

## 2-1. 세 데이터 inner로 결합

In [33]:
merge_all = pd.merge(merge_p_cont, pmnt_pol, on=['JOIN_SN', 'POL_SN'], how='inner').sort_values('POL_SN')
merge_all

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
0      378480     1616              C       1   46          4              3   
1       35557     1663              C       2   50          5              2   
2      705387     1872              C       1   34          3              2   
3      477431     4948              C       1   70          7              3   
4     1948230     5142              C       1   36          3              2   
...       ...      ...            ...     ...  ...        ...            ...   
6225  1082607  9989189              C       2   51          5              1   
6226  1212457  9991015              C       2   37          3              3   
6227  1365296  9992652              C       1   52          5              8   
6228  1324915  9993452              C       2   40          4              2   
6229  1373416  9994891              C       2   30          3              2   

      reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
0                  1          1           3         96               2   
1                  1          1           2        210               2   
2                  1          1           3         65               2   
3                  1          1           4        140               3   
4                  1          1           1         53               2   
...              ...        ...         ...        ...             ...   
6225               1          1           1        300               2   
6226               1          1           1         69               2   
6227               1          1           6         49               2   
6228               1          1           1         83               2   
6229               1          1           1         88               2   

      CT_PY_PD  mean_AMT   month_AMT   z_score  m_z_score  GBN_CD_1  GBN_CD_2  \
0           20      90.0   90.000000 -0.021843  -0.020172         1         1   
1           20     210.0  210.000000  0.011177   0.013003         1         1   
2           20      64.0   64.000000 -0.028997  -0.027359         1         1   
3            1     140.0   11.666667 -0.008085  -0.041827         1         1   
4           20      53.0   53.000000 -0.032024  -0.030400         1         1   
...        ...       ...         ...       ...        ...       ...       ...   
6225        15     300.0  300.000000  0.035942   0.037884         1         1   
6226        20      68.0   68.000000 -0.027896  -0.026254         1         1   
6227         5      49.0   49.000000 -0.033124  -0.031506         0         0   
6228        30      74.5   74.500000 -0.026108  -0.024457         1         1   
6229        20      85.5   85.500000 -0.023081  -0.021416         1         1   

      GBN_CD_3  GBN_CD_4 INS_GBN_CD  총지급금액  총지급 건수  기타  배책  사망/후유  수술/입원  실손  \
0            1         0          c   2026       9   0   0      0      0   9   
1            1         0          b   5930      12   0   0      0      6   6   
2            1         1          b     77       1   0   0      0      0   1   
3            1         0          c     17       1   0   0      0      0   1   
4            1         0          b     63       1   0   0      0      1   0   
...        ...       ...        ...    ...     ...  ..  ..    ...    ...  ..   
6225         1         0          h    949       3   0   0      0      3   0   
6226         1         1          c     35       1   0   0      0      0   1   
6227         0         1          h   1012       2   1   0      0      0   0   
6228         1         0          b    460       1   0   0      0      0   1   
6229         1         0          b     80       1   0   0      0      1   0   

      자동차  진단/통원  화재/재물  
0       0      0      0  
1       0      0      0  
2       0      0      0  
3       0      0      0  
4       0      0      0  
...   ...    ...    ...  
6225    0      0      0  
6226    0     

## 2-2. 보험금 지급이 없는 데이터는 모두 0으로 진행
  * 결합된 테이블 : merge_all3

In [34]:
merge_all2 = pd.merge(merge_p_cont, pmnt_pol, on=['JOIN_SN', 'POL_SN'], how='left').sort_values('POL_SN')
merge_all2

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
0       902434        8              C       1   29          2              5   
1      1667175      111              C       2   42          4              1   
2       594950      179              C       1   45          4              1   
3      1388667      722              C       2   20          2              3   
4       453567     1031              C       2   54          5              2   
...        ...      ...            ...     ...  ...        ...            ...   
58422   242990  9996700              C       2   30          3              1   
58423  1697168  9996939              C       1   23          2              3   
58424   314074  9997065              C       1   39          3              1   
58425  1406609  9997129              C       2   46          4              3   
58426  1494449  9997274              C       2   55          5              2   

       reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
0                   1          1           1         35               2   
1                   1          1           3         60               2   
2                   1          1           1        330               2   
3                   1          1           1         28               2   
4                   1          1           6         29               2   
...               ...        ...         ...        ...             ...   
58422               1          1           1         47               2   
58423               1          1           6         25               2   
58424               1          1           1        190               2   
58425               1          1           1         24               2   
58426               1          1           6         34               2   

       CT_PY_PD    mean_AMT   month_AMT   z_score  m_z_score  GBN_CD_1  \
0            15   35.000000   35.000000 -0.036977  -0.035377         1   
1            25   60.000000   60.000000 -0.030098  -0.028465         1   
2            20  330.000000  330.000000  0.044197   0.046178         1   
3            30   28.000000   28.000000 -0.038903  -0.037312         1   
4            20   29.000000   29.000000 -0.038628  -0.037035         0   
...         ...         ...         ...       ...        ...       ...   
58422        20   45.666667   45.666667 -0.034042  -0.032428         1   
58423        25   25.000000   25.000000 -0.039728  -0.038141         1   
58424        20  190.000000  190.000000  0.005674   0.007474         1   
58425         5   24.000000   24.000000 -0.040004  -0.038418         1   
58426        20   34.000000   34.000000 -0.037252  -0.035653         0   

       GBN_CD_2  GBN_CD_3  GBN_CD_4 INS_GBN_CD  총지급금액  총지급 건수  기타  배책  사망/후유  \
0             1         1         1        NaN    NaN     NaN NaN NaN    NaN   
1             1         1         0        NaN    NaN     NaN NaN NaN    NaN   
2             1         1         0        NaN    NaN     NaN NaN NaN    NaN   
3             1         1         1        NaN    NaN     NaN NaN NaN    NaN   
4             1         1         0        NaN    NaN     NaN NaN NaN    NaN   
...         ...       ...       ...        ...    ...     ...  ..  ..    ...   
58422         1         1         0        NaN    NaN     NaN NaN NaN    NaN   
58423         1         0         0        NaN    NaN     NaN NaN NaN    NaN   
58424         1         1         0        NaN    NaN     NaN NaN NaN    NaN   
58425         1         0         1        NaN    NaN     NaN NaN NaN    NaN   
58426         1         0         0        NaN    NaN     NaN NaN NaN    NaN   

       수술/입원  실손  자동차  진단/통원  화재/재물  
0        NaN NaN  NaN    NaN    NaN  
1        NaN NaN  NaN    NaN    NaN  
2        NaN NaN  NaN    NaN    NaN  
3        NaN NaN  NaN    NaN    NaN  
4        NaN NaN  NaN    NaN    NaN  
...      ...  ..  ...    ...    ...  
58422    NaN NaN 

In [19]:
merge_all2.isnull().sum()

JOIN_SN               0
POL_SN                0
CT_RLPS_GBN_CD        0
GENDER                0
AGE                   0
AGE_GROUP             0
reduce_GBN_CD         0
reduce_STAT_CD        0
YM_202006             0
CT_CHNL_CD            0
CT_PY_AMT             0
CT_PY_CYCLE_CD        0
CT_PY_PD              0
mean_AMT              0
month_AMT             0
z_score               0
m_z_score             0
GBN_CD_1              0
GBN_CD_2              0
GBN_CD_3              0
GBN_CD_4              0
INS_GBN_CD        52199
총지급금액             52199
총지급 건수            52199
기타                52199
배책                52199
사망/후유             52199
수술/입원             52199
실손                52199
자동차               52199
진단/통원             52199
화재/재물             52199
dtype: int64

In [35]:
merge_all3 = merge_all2.drop(columns=['INS_GBN_CD'])
merge_all3

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
0       902434        8              C       1   29          2              5   
1      1667175      111              C       2   42          4              1   
2       594950      179              C       1   45          4              1   
3      1388667      722              C       2   20          2              3   
4       453567     1031              C       2   54          5              2   
...        ...      ...            ...     ...  ...        ...            ...   
58422   242990  9996700              C       2   30          3              1   
58423  1697168  9996939              C       1   23          2              3   
58424   314074  9997065              C       1   39          3              1   
58425  1406609  9997129              C       2   46          4              3   
58426  1494449  9997274              C       2   55          5              2   

       reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
0                   1          1           1         35               2   
1                   1          1           3         60               2   
2                   1          1           1        330               2   
3                   1          1           1         28               2   
4                   1          1           6         29               2   
...               ...        ...         ...        ...             ...   
58422               1          1           1         47               2   
58423               1          1           6         25               2   
58424               1          1           1        190               2   
58425               1          1           1         24               2   
58426               1          1           6         34               2   

       CT_PY_PD    mean_AMT   month_AMT   z_score  m_z_score  GBN_CD_1  \
0            15   35.000000   35.000000 -0.036977  -0.035377         1   
1            25   60.000000   60.000000 -0.030098  -0.028465         1   
2            20  330.000000  330.000000  0.044197   0.046178         1   
3            30   28.000000   28.000000 -0.038903  -0.037312         1   
4            20   29.000000   29.000000 -0.038628  -0.037035         0   
...         ...         ...         ...       ...        ...       ...   
58422        20   45.666667   45.666667 -0.034042  -0.032428         1   
58423        25   25.000000   25.000000 -0.039728  -0.038141         1   
58424        20  190.000000  190.000000  0.005674   0.007474         1   
58425         5   24.000000   24.000000 -0.040004  -0.038418         1   
58426        20   34.000000   34.000000 -0.037252  -0.035653         0   

       GBN_CD_2  GBN_CD_3  GBN_CD_4  총지급금액  총지급 건수  기타  배책  사망/후유  수술/입원  실손  \
0             1         1         1    NaN     NaN NaN NaN    NaN    NaN NaN   
1             1         1         0    NaN     NaN NaN NaN    NaN    NaN NaN   
2             1         1         0    NaN     NaN NaN NaN    NaN    NaN NaN   
3             1         1         1    NaN     NaN NaN NaN    NaN    NaN NaN   
4             1         1         0    NaN     NaN NaN NaN    NaN    NaN NaN   
...         ...       ...       ...    ...     ...  ..  ..    ...    ...  ..   
58422         1         1         0    NaN     NaN NaN NaN    NaN    NaN NaN   
58423         1         0         0    NaN     NaN NaN NaN    NaN    NaN NaN   
58424         1         1         0    NaN     NaN NaN NaN    NaN    NaN NaN   
58425         1         0         1    NaN     NaN NaN NaN    NaN    NaN NaN   
58426         1         0         0    NaN     NaN NaN NaN    NaN    NaN NaN   

       자동차  진단/통원  화재/재물  
0      NaN    NaN    NaN  
1      NaN    NaN    NaN  
2      NaN    NaN    NaN  
3      NaN    NaN    NaN  
4      NaN    NaN    NaN  
...    ...    ...    ...  
58422  NaN    NaN    NaN  
58423  NaN    NaN    NaN  
58424  NaN    NaN    NaN  
58425  NaN   

In [36]:
merge_all3.fillna(0, inplace=True)
merge_all3

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
0       902434        8              C       1   29          2              5   
1      1667175      111              C       2   42          4              1   
2       594950      179              C       1   45          4              1   
3      1388667      722              C       2   20          2              3   
4       453567     1031              C       2   54          5              2   
...        ...      ...            ...     ...  ...        ...            ...   
58422   242990  9996700              C       2   30          3              1   
58423  1697168  9996939              C       1   23          2              3   
58424   314074  9997065              C       1   39          3              1   
58425  1406609  9997129              C       2   46          4              3   
58426  1494449  9997274              C       2   55          5              2   

       reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
0                   1          1           1         35               2   
1                   1          1           3         60               2   
2                   1          1           1        330               2   
3                   1          1           1         28               2   
4                   1          1           6         29               2   
...               ...        ...         ...        ...             ...   
58422               1          1           1         47               2   
58423               1          1           6         25               2   
58424               1          1           1        190               2   
58425               1          1           1         24               2   
58426               1          1           6         34               2   

       CT_PY_PD    mean_AMT   month_AMT   z_score  m_z_score  GBN_CD_1  \
0            15   35.000000   35.000000 -0.036977  -0.035377         1   
1            25   60.000000   60.000000 -0.030098  -0.028465         1   
2            20  330.000000  330.000000  0.044197   0.046178         1   
3            30   28.000000   28.000000 -0.038903  -0.037312         1   
4            20   29.000000   29.000000 -0.038628  -0.037035         0   
...         ...         ...         ...       ...        ...       ...   
58422        20   45.666667   45.666667 -0.034042  -0.032428         1   
58423        25   25.000000   25.000000 -0.039728  -0.038141         1   
58424        20  190.000000  190.000000  0.005674   0.007474         1   
58425         5   24.000000   24.000000 -0.040004  -0.038418         1   
58426        20   34.000000   34.000000 -0.037252  -0.035653         0   

       GBN_CD_2  GBN_CD_3  GBN_CD_4  총지급금액  총지급 건수   기타   배책  사망/후유  수술/입원  \
0             1         1         1    0.0     0.0  0.0  0.0    0.0    0.0   
1             1         1         0    0.0     0.0  0.0  0.0    0.0    0.0   
2             1         1         0    0.0     0.0  0.0  0.0    0.0    0.0   
3             1         1         1    0.0     0.0  0.0  0.0    0.0    0.0   
4             1         1         0    0.0     0.0  0.0  0.0    0.0    0.0   
...         ...       ...       ...    ...     ...  ...  ...    ...    ...   
58422         1         1         0    0.0     0.0  0.0  0.0    0.0    0.0   
58423         1         0         0    0.0     0.0  0.0  0.0    0.0    0.0   
58424         1         1         0    0.0     0.0  0.0  0.0    0.0    0.0   
58425         1         0         1    0.0     0.0  0.0  0.0    0.0    0.0   
58426         1         0         0    0.0     0.0  0.0  0.0    0.0    0.0   

        실손  자동차  진단/통원  화재/재물  
0      0.0  0.0    0.0    0.0  
1      0.0  0.0    0.0    0.0  
2      0.0  0.0    0.0    0.0  
3      0.0  0.0    0.0    0.0  
4      0.0  0.0    0.0    0.0  
...    ...  ...    ...    ...  
58422  0.0  0.0    0.0    0.0  
58423  0.0  0.0    0.0    0.0  
58424  0.0  0.0    

In [39]:
to_int = merge_all3.columns[-10:]
to_int

Index(['총지급금액', '총지급 건수', '기타', '배책', '사망/후유', '수술/입원', '실손', '자동차', '진단/통원',
       '화재/재물'],
      dtype='object')

In [40]:
merge_all3[to_int] = merge_all3[to_int].astype(int)
merge_all3

JOIN_SN   POL_SN CT_RLPS_GBN_CD  GENDER  AGE  AGE_GROUP  reduce_GBN_CD  \
0       902434        8              C       1   29          2              5   
1      1667175      111              C       2   42          4              1   
2       594950      179              C       1   45          4              1   
3      1388667      722              C       2   20          2              3   
4       453567     1031              C       2   54          5              2   
...        ...      ...            ...     ...  ...        ...            ...   
58422   242990  9996700              C       2   30          3              1   
58423  1697168  9996939              C       1   23          2              3   
58424   314074  9997065              C       1   39          3              1   
58425  1406609  9997129              C       2   46          4              3   
58426  1494449  9997274              C       2   55          5              2   

       reduce_STAT_CD  YM_202006  CT_CHNL_CD  CT_PY_AMT  CT_PY_CYCLE_CD  \
0                   1          1           1         35               2   
1                   1          1           3         60               2   
2                   1          1           1        330               2   
3                   1          1           1         28               2   
4                   1          1           6         29               2   
...               ...        ...         ...        ...             ...   
58422               1          1           1         47               2   
58423               1          1           6         25               2   
58424               1          1           1        190               2   
58425               1          1           1         24               2   
58426               1          1           6         34               2   

       CT_PY_PD    mean_AMT   month_AMT   z_score  m_z_score  GBN_CD_1  \
0            15   35.000000   35.000000 -0.036977  -0.035377         1   
1            25   60.000000   60.000000 -0.030098  -0.028465         1   
2            20  330.000000  330.000000  0.044197   0.046178         1   
3            30   28.000000   28.000000 -0.038903  -0.037312         1   
4            20   29.000000   29.000000 -0.038628  -0.037035         0   
...         ...         ...         ...       ...        ...       ...   
58422        20   45.666667   45.666667 -0.034042  -0.032428         1   
58423        25   25.000000   25.000000 -0.039728  -0.038141         1   
58424        20  190.000000  190.000000  0.005674   0.007474         1   
58425         5   24.000000   24.000000 -0.040004  -0.038418         1   
58426        20   34.000000   34.000000 -0.037252  -0.035653         0   

       GBN_CD_2  GBN_CD_3  GBN_CD_4  총지급금액  총지급 건수  기타  배책  사망/후유  수술/입원  실손  \
0             1         1         1      0       0   0   0      0      0   0   
1             1         1         0      0       0   0   0      0      0   0   
2             1         1         0      0       0   0   0      0      0   0   
3             1         1         1      0       0   0   0      0      0   0   
4             1         1         0      0       0   0   0      0      0   0   
...         ...       ...       ...    ...     ...  ..  ..    ...    ...  ..   
58422         1         1         0      0       0   0   0      0      0   0   
58423         1         0         0      0       0   0   0      0      0   0   
58424         1         1         0      0       0   0   0      0      0   0   
58425         1         0         1      0       0   0   0      0      0   0   
58426         1         0         0      0       0   0   0      0      0   0   

       자동차  진단/통원  화재/재물  
0        0      0      0  
1        0      0      0  
2        0      0      0  
3        0      0      0  
4        0      0      0  
...    ...    ...    ...  
58422    0      0      0  
58423    0      0      0  
58424    0      0      0  
58425    0   

In [41]:
merge_all3.isnull().sum()

JOIN_SN           0
POL_SN            0
CT_RLPS_GBN_CD    0
GENDER            0
AGE               0
AGE_GROUP         0
reduce_GBN_CD     0
reduce_STAT_CD    0
YM_202006         0
CT_CHNL_CD        0
CT_PY_AMT         0
CT_PY_CYCLE_CD    0
CT_PY_PD          0
mean_AMT          0
month_AMT         0
z_score           0
m_z_score         0
GBN_CD_1          0
GBN_CD_2          0
GBN_CD_3          0
GBN_CD_4          0
총지급금액             0
총지급 건수            0
기타                0
배책                0
사망/후유             0
수술/입원             0
실손                0
자동차               0
진단/통원             0
화재/재물             0
dtype: int64

In [42]:
merge_all3.to_csv('merge_all.csv', index=False, encoding='utf-8-sig')